client.go

```diff
@@ -141,23 +155,29 @@ func (c Client) RawSecret(name string) (data []byte, err error) {
 	data, err = ioutil.ReadAll(resp.Body)
 	if err != nil {
 		c.Errorf("Error reading response body for secret %v: %v", name, err)
+		c.failCountInc()
 		return nil, err
 	}
 
 	switch resp.StatusCode {
 	case 200:
+		c.failCountClear()
+		c.lastSuccessUpdate()
```

Comment:
We always call these two together.  Maybe they should be one function.  c.MarkSuccess() or something.